src/librustc_typeck/check/method/suggest.rs

```diff
@@ -173,7 +173,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                      if mode == Mode::MethodCall {
                                          "method"
                                      } else {
-                                         "associated item"
+                                         match item_name.as_str().chars().next() {
+                                             Some(name) => {
+                                                 if name.is_lowercase() {
+                                                     "function or associated item"
```

Comment:
Why can't it be a const?

I wasn't sure if we call these methods, opinions seem to vary.

I put "function" first because newbies don't know what "associated blah" means.